nemo/lightning/run/plugins.py

```diff
@@ -177,3 +176,68 @@ def setup(self, task: run.Partial | run.Script, executor: run.Executor):
             logging.warning(
                 f"The {self.__class__.__name__} will have no effect as WANDB_API_KEY environment variable is not set."
             )
+
+
+@dataclass(kw_only=True)
+class ConfigValidationPlugin(run.Plugin):
+    """
+    A plugin for validating a NeMo task with its executor.
```

Comment:
suggestion: if `run.Config[T]` were constrained to Pydantic BaseModels, I think we would get validation for free.

Reply:
after a second look, I dont think thats the purpose of what this PR accomplishes.